### PR TITLE
Use linux-musl runtime on Alpine Helix workers

### LINKED ISSUE #36825: runtime-coreclr outerloop pipeline is broken
Since https://github.com/dotnet/runtime/pull/35426, the runtime-coreclr outerloop pipeline has been failing since it hasn't been able to find a `dotnet` to execute with.

cc: @jashook @trylek @nattress

### PR DESCRIPTION
**Pull request by nattress**
Fixes https://github.com/dotnet/runtime/issues/36825

### CODE REVIEW

**Review by jkoritzinsky — APPROVED** [review]

**Review by jashook — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
Link to an outerloop build: https://dev.azure.com/dnceng/public/_build/results?buildId=655809&view=results

**Comment by jkoritzinsky** [comment]
Formatting job failed to download clang-format

**Review by trylek — APPROVED** [review]
Looks just as beautiful as Jeremy's version, thanks for fixing!

**Review by jkoritzinsky — CHANGES_REQUESTED** [review]
Outerloop job is still failing with the same issue. Apparently TargetOS is getting passed in as Linux when we send the tests to Helix.

**Comment by jkoritzinsky** [comment]
We need to change the following line to also account for the `osSubGroup` parameter (which I believe will also need to be added to the parameter list at the top of the YAML file): https://github.com/dotnet/runtime/blob/a2ec75f3feaabec604fb03ad5252c92d3e152aa9/eng/pipelines/common/templates/runtimes/send-to-helix-step.yml#L42

**Comment by nattress** [comment]
Do we need to update the thing that invokes this YAML to also pass in `osSubGroup` or will it just work?

**Comment by trylek** [comment]
I'm afraid that the former is true, there's nothing about YAML that just works...

**Comment by trylek** [comment]
https://github.com/dotnet/runtime/blob/a2ec75f3feaabec604fb03ad5252c92d3e152aa9/eng/pipelines/common/templates/runtimes/run-test-job.yml#L283

File: eng/pipelines/common/templates/runtimes/send-to-helix-step.yml [hunk 1]
```diff
@@ -82,7 +82,7 @@ steps:
       condition: ${{ parameters.condition }}
     env:
       __BuildArch: ${{ parameters.archType }}
-      __TargetOS: ${{ parameters.osGroup }}${{ parameters.osSubGroup }}
+      __TargetOS: ${{ parameters.osGroup }}${{ parameters.osSubgroup }}
```
Comment (trylek) [inline eng/pipelines/common/templates/runtimes/send-to-helix-step.yml:85]: Looks good. At some point we may want to consider splitting these too into something like __TargetOSGroup / Subgroup but please avoid doing that as part of this change :-).

**Comment by nattress** [comment]
Thanks Tomas 😄 Btw, when I fixed my osSubgroup casing, I grepped the repo and found one other place we capitalize it differently: https://github.com/dotnet/runtime/blob/bdc09433b6d451e66943de959afdab92117c502e/eng/pipelines/libraries/base-job.yml#L65

My understanding is YAML is case sensitive so this is probably not working right.

**Comment by trylek** [comment]
YAML is mostly case insensitive (not only when resolving variable names, even when comparing strings) so this should be technically fine but I don't deny I find the Pascal casing of SubGroup just as annoying as in case of RunTime. Please feel free to fix that one as part of this change if you have the energy to do so.

**Review by jkoritzinsky — APPROVED** [review]

**Pull request merged**